lib/salus/report.rb

```diff
@@ -222,9 +224,26 @@ def publish_report(directive)
       end
     end
 
+    def satisfies_filter?(directive)
+      return true if @report_filter == 'all'
+      return false if @report_filter == 'none'
+
+      # Use regex to capture two parts of the filter, delimited by a ':'
+      filter_regex = Regexp.new('([a-zA-Z0-9]+):?([a-zA-Z0-9\*]*)')
+      captured_parameters = filter_regex.match(@report_filter)
+      directive_key = captured_parameters[1]
+      directive_value = directive[directive_key]
+      # Since the value of the second capture group can follow a regex
+      # pattern (i.e. can use wildcards), we build a regex from it
+      # and use it to match against the directive value
+      directive_value_regex = Regexp.new(captured_parameters[2])
+
+      directive.key?(directive_key) && directive_value_regex.match?(directive_value)
```

Comment:
Not all keys will appear as part of directive.
For example, this is part of the SECT ticket, but `name` will never appear as part of the `report section`.
```
--reports name:* # run all reports
```

Report sections look like below, with no `name` keyword.
```yaml
reports:
  - uri: file://out.json
    format: json
  - uri: file://sarif.json
    format: sarif
```

Reply:
@joshua-ostrom How did you want the code to work with `name`? What are the possible values for `name`?  Only `*`?  

Note the ticket says
```
--reports none # don't run reports
--reports name:* # run all reports
```

Should that instead be
```
--reports none # don't run reports
--reports * # run all reports
```

Reply:
```reports:
  - uri: file://out.json
    format: json
    name: arcus
  - uri: file://sarif.json
    format: sarif
    name: vuln_inventory
  - uri: file://sarif2.json
    format: sarif

```

none # filter away all reports, don't create any reports.  
name:* # Run all reports (same as not specifying --reports)
name: arcus # run only the arcus report
name:arcus vuln_inventory # run both of these reports

By using --reports key:value_filter the thought was we could support filtering of multiple fields (for example format and name initially but URI etc in the future).  That said, I'm not tied to that exact format there's latitude if we want to change it up.

* and none are just there for convenience.  We can probably drop * if needed (as the user can just not use --reports CLI in that case).  None would be the same as --reports name:foo (assuming there is no report named foo).

Reply:
@joshua-ostrom @ghbren I added an `all` value in place of `--reports name:*` since it mirrors `none`. Using `all` is just like not specifying the flag at all.